### PR TITLE
Add toggle for advanced table name humanization

### LINKED ISSUE #2660: New display_name logic capitialized "reciever_id" as "Recieve R ID"
Working on a new "Avian Singles" test database and saw this 

screen shot 2016-05-24 at 12 29 02 pm

### PR DESCRIPTION
**Pull request by camsaul**
Hitting the toggle will also toggle between humanizing existing tables using the older "simple" humanization (only split on hypens, spaces and dashes) and the "advanced" one introduced in 0.17. Toggle is pictured here:

screen shot 2016-07-25 at 6 18 42 pm

(typo already fixed)

Being a `Setting`, this option can also be set via the env var `MB_ENABLE_ADVANCED_HUMANIZATION`. 

Tests included! 

Fixes #2660

### CODE REVIEW

**Comment by camsaul** [comment]
tests included! 😻

**Comment by camsaul** [comment]
does anyone want to review 👍

File: frontend/src/metabase/admin/settings/selectors.js [hunk 1]
```diff
@@ -41,6 +41,11 @@ const SECTIONS = [
                 key: "anon-tracking-enabled",
                 display_name: "Anonymous Tracking",
                 type: "boolean"
+            },
+            {
+                key: "enable-advanced-humanization",
+                display_name: "Friendly Table Names",
```
Comment (tlrobinson) [inline frontend/src/metabase/admin/settings/selectors.js]: "Table and Field Names", right?

**Comment by tlrobinson** [comment]
Two problems:

1. This is supposed to apply to field names as well.
2. Currently this blows away any changes the user made to table names without any sort of confirmation from the admin. I think we need to do one or both of the following:
  * Prompt the admin to confirm that they would like to reset every table and field name in every database.
  * Only reset the table/field names that haven't been changed the the defaults. We could do this by comparing the saved name to what would have been generated by the previous setting (though this will break in some cases if we change the logic in the future).

**Comment by salsakran** [comment]
Yeah, this doesn't really solve the entire problem (field names are an even bigger problem as there are more of them to correct).

I agree w/ Tom that we should only change back table names that were unmodified.

**Comment by camsaul** [comment]
good points. I will make those changes

**Comment by camsaul** [comment]
ok, tweaked logic and added additional tests.

**Comment by tlrobinson** [comment]
Did we want a modal confirmation?

**Comment by camsaul** [comment]
i tweaked it so it doesn't change custom names.

**Pull request merged**